### PR TITLE
add tests for the functions that set and/or depend on global variables

### PR DESCRIPTION
**Pull request by cjyetman**

### CODE REVIEW

File: tests/testthat/test-set_analysis_inputs_path.R [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      func = function(set_analysis_inputs_path, data_location_ext) {
+        set_analysis_inputs_path(data_location_ext)
+      },
+      args = list(set_analysis_inputs_path, data_location_ext)
+    )
+
+  expect_equal(output, data_location_ext)
+})
+
+test_that("sets appropriate global variables", {
```
Comment (jdhoffa) [inline tests/testthat/test-set_analysis_inputs_path.R]: can you be more specific with the name of this test? I am trying to figure out what it is expecting to happen here

File: tests/testthat/test-set_analysis_inputs_path.R [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        set_analysis_inputs_path,
+        data_location_ext,
+        proj_data_location_ext
+      )
+    )
+
+  expect_equal(output, proj_data_location_ext)
+})
+
+test_that("sets appropriate global variables", {
```
Comment (jdhoffa) [inline tests/testthat/test-set_analysis_inputs_path.R]: same as above, what behaviour is this testing for that the previous test isn't? it's pretty difficult for me to understand what is going on here

**Review by jdhoffa — DISMISSED** [review]
NB comments about being more specific about test names (i was a bit confused at what the differences were between the tests or why it was necessary to have multiple)

but functionally lgtm

File: tests/testthat/test-set_analysis_inputs_path.R (hunk 1 above)
Comment (cjyetman) [reply tests/testthat/test-set_analysis_inputs_path.R]:
```suggestion
test_that("returns expected value", {
```
These are wrong and were carried over from another test. Is this better?

Comment (cjyetman) [reply tests/testthat/test-set_analysis_inputs_path.R]: or better... 
```suggestion
test_that("returns expected value when overridden by a global variable `proj_data_location_ext`", {
```

File: tests/testthat/test-set_analysis_inputs_path.R (hunk 2 above)
Comment (cjyetman) [reply tests/testthat/test-set_analysis_inputs_path.R]:
```suggestion
test_that("returns expected value when overridden by a global variable `port_holdings_date`", {
```
These are wrong and were carried over from another test. Is this better?

Comment (jdhoffa) [reply tests/testthat/test-set_analysis_inputs_path.R:37]: The point is that there are two different tests with the exact same name, so it's pretty difficult to understand what the difference is that they are testing?

The suggestion you gave is still the same for both which would still be confusing

File: tests/testthat/test-set_analysis_inputs_path.R [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
```
Comment (cjyetman) [inline tests/testthat/test-set_analysis_inputs_path.R]:
```suggestion
test_that("returns expected value when not overridden by a global variable", {
```

File: tests/testthat/test-set_analysis_inputs_path.R [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        set_analysis_inputs_path,
+        data_location_ext,
+        proj_data_location_ext
+      )
+    )
+
+  expect_equal(output, proj_data_location_ext)
+})
+
+test_that("returns expected value when overridden by a global variable", {
```
Comment (cjyetman) [inline tests/testthat/test-set_analysis_inputs_path.R]:
```suggestion
test_that("returns expected value when overridden by a global variable `port_holdings_date`", {
```

**Pull request merged**